### PR TITLE
Ssh backend tests

### PR DESCRIPTION
**Pull request by arlimus**

### CODE REVIEW

File: lib/vulcano/backend/specinfra.rb [hunk 1]
```diff
@@ -248,7 +251,7 @@ def link_path
 
       def content
         s = Specinfra::Runner.get_file_content(@path).stdout
-        if s.empty? && (directory? or size.nil? or size > 0)
+        if s.empty? && (directory? or (size || 0) > 0 or !exist?)
```
Comment (chris-rock) [inline lib/vulcano/backend/specinfra.rb]: if size is nil, this becomes 0 and is not greater than 0, is this intended?

File: test/runner/cookbooks/test/recipes/default.rb [hunk 2]
```diff
@@ -32,13 +41,32 @@
...
 
-execute 'run tests' do
-  command '/opt/chef/embedded/bin/ruby -I lib test/runner/test.rb test/runner/tests/*_test.rb'
+execute 'run local tests' do
+  command '/opt/chef/embedded/bin/ruby -I lib test/runner/test_local.rb test/runner/tests/*_test.rb'
+  cwd '/tmp/kitchen/data'
+end
+
+execute 'run ssh tests' do
+  command '/opt/chef/embedded/bin/ruby -I lib test/runner/test_ssh.rb test/runner/tests/*_test.rb'
```
Comment (chris-rock) [inline test/runner/cookbooks/test/recipes/default.rb:70]: I like that we can run ssh tests like local tests, yeah!

File: lib/vulcano/backend/specinfra.rb (hunk 1 above)
Comment (arlimus) [inline lib/vulcano/backend/specinfra.rb]: Yes it is, it's a way of not calling size twice, as the result isn't cached and instead called twice... However, it looks confusing, let me take another look.

**Comment by chris-rock** [comment]
Thanks @arlimus

**Pull request merged**